toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"strings"
+	"path/filepath"
+	"io/ioutil"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func enableVerityPartition(Verity imagecustomizerapi.Verity, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// Ensure the VerityPartition value is one of the supported types: 'root' or 'user'.
+	if Verity.VerityTab != "root" && Verity.VerityTab != "user" {
+		return fmt.Errorf("invalid VerityPartition: %s. It should be either 'root' or 'user'", Verity.VerityTab)
+	}
+
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	systemdVerityDracutModule := "systemd-veritysetup"
+	err = buildDracutModule(systemdVerityDracutModule, imageChroot) 
+	if err != nil {
+		return fmt.Errorf("failed to integrate systemd veritysetup dracut module: %w", err)
+	}
+
+	// Update mariner config file with the new generated initramfs file.
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to update mariner config file: %w", err)
+	}
+
+	// Calculate the root hash using veritysetup
+	rootHash, err := calculateRootHash(Verity, imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to calculate the root hash: %w", err)
+	}
+
+	// Update grub.cfg with the new kernel command-line arguments.
+	err = updateGrubConfigWithVerityArgs(rootHash, Verity, imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to update grub.cfg with verity arguments: %w", err)
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	buildDracutModuleArgs := []string{
+		"dracut", "-f", "-a",
```

Comment:
Calling `dracut` directly can cause a lot of problems. The correct way to handle this is to add the module by creating a new config file under `/etc/dracut.conf.d/`. After that, reinstalling the `dracut` package should trigger the update to the initramfs file.